### PR TITLE
docs: delete lspconfig-keybindings section

### PR DESCRIPTION
**Pull request by gpanders**
This section is way too verbose and encourages sloppy copy-pasting (not to mention it is out of date). Nvim already ships help documentation for configuring the LSP client, so instead of duplicating that information, point users toward upstream docs instead.

Fixes: https://github.com/neovim/nvim-lspconfig/issues/3174

Related: https://github.com/neovim/nvim-lspconfig/pull/3122

### CODE REVIEW

File: doc/lspconfig.txt [hunk 1]
```diff
@@ -169,10 +169,11 @@ passed overrides to `setup {}` are:
   Callback invoked by Nvim's built-in client when attaching a buffer to a
   language server. Often used to set Nvim (buffer or global) options or to
   override the Nvim client properties (`server_capabilities`) after a
-  language server attaches. Most commonly used for settings buffer
-  local keybindings. See |lspconfig-keybindings| for a usage example.
+  language server attaches.
 
-- {settings} `table <string, string|table|bool>`  
+  Prefer using an |LspAttach| autocommand handler instead.
```
Comment (dundargoc) [inline doc/lspconfig.txt:174]: README mentions `on_attach` instead of LspAttach, might be worth changing it there.

File: doc/lspconfig.txt [hunk 2]
```diff
@@ -473,64 +474,6 @@ contained in `:LspInfo`:
...
-      vim.keymap.set('n', '<space>wl', function()
-        print(vim.inspect(vim.lsp.buf.list_workspace_folders()))
-      end, opts)
-      vim.keymap.set('n', '<space>D', vim.lsp.buf.type_definition, opts)
-      vim.keymap.set('n', '<space>rn', vim.lsp.buf.rename, opts)
-      vim.keymap.set({ 'n', 'v' }, '<space>ca', vim.lsp.buf.code_action, opts)
-      vim.keymap.set('n', 'gr', vim.lsp.buf.references, opts)
-      vim.keymap.set('n', '<space>f', function()
-        vim.lsp.buf.format { async = true }
-      end, opts)
```
Comment (dundargoc) [inline doc/lspconfig.txt:530]: thought: wonder if this is worth upstreaming to core. Doesn't have to be `<space>f`, but something for formatting.

**Review by dundargoc — APPROVED** [review]
praise: well done

File: doc/lspconfig.txt (hunk 2 above)
Comment (gpanders) [reply doc/lspconfig.txt:530]: We don't need it. `'formatexpr'` is already set, so `gq` "just works".

A default key mapping for "format the whole buffer" might be useful though, but that's adjacent to LSP.

File: doc/lspconfig.txt (hunk 1 above)
Comment (gpanders) [reply doc/lspconfig.txt:174]: Done

**Review by justinmk — APPROVED** [review]
💯

**Comment by alexvitkov** [comment]
This is a bad decision - the keybinds section provided a great jumping-off point for 99% of users, and over time has become the de facto resource for bootstrapping a new configuration - when you google something like "neovim lsp go to definition" virtually all resources lead back to this README, which has now been stripped out of any useful content.

**Comment by dundargoc** [comment]
Pretty much all keybindings have been upstreamed to neovim and is the default now.

**Comment by josalvatorre** [comment]
>Pretty much all keybindings have been upstreamed to neovim and is the default now.

@dundargoc Could you please point to where those documented keybindings are? It's not obvious to a newcomer like me where those should be. I didn't find them in https://neovim.io/doc/user/lsp.html#_defaults. Some of the bindings aren't working for me out of the box. I'd like to know which ones I need to add manually.

**Comment by justinmk** [comment]
>  I didn't find them in https://neovim.io/doc/user/lsp.html#_defaults. Some of the bindings aren't working for me out of the box.

Always mention the exact Nvim version you are using. And try the latest release, if you haven't already.

**Pull request merged**